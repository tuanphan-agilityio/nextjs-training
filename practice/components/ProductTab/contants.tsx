const TAB_ITEMS = [
  {
    id: '1',
    label: 'Description',
    content: (
      <div className='text-md sm:text-sm'>
        <p className='text-secondary font-primary-bold'>Detail Product</p>
        <ul className='mt-5 ml-6 list-disc'>
          {Array(3)
            .fill('Lorem ipsum dolor sit amet, consectetur adipiscing elit')
            .map((item, index) => (
              <li key={index} className='text-secondary'>
                {item}
              </li>
            ))}
        </ul>
      </div>
    ),
  },
  {
    id: '2',
    label: 'Review (5)',
    content: <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit</p>,
  },
];

export { TAB_ITEMS };
