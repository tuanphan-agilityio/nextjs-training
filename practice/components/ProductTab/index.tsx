import { FC } from 'react';

import Tabs from '@/components/Tabs';

import { TAB_ITEMS } from './contants';

const ProductTab: FC = () => {
  return (
    <div className='container'>
      <Tabs items={TAB_ITEMS} />
    </div>
  );
};

export default ProductTab;
