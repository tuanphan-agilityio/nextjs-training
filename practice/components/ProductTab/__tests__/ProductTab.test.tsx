import { render } from '@testing-library/react';

import ProductTab from '..';

describe('ProductTab Component', () => {
  it('should match the snapshot', () => {
    const { container } = render(<ProductTab />);
    expect(container).toMatchSnapshot();
  });
});
