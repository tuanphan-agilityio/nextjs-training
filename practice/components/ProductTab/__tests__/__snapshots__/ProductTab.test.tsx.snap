// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTab Component should match the snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <div>
      <ul
        class="flex justify-center gap-10 sm:gap-4"
      >
        <li
          class="pb-2 font-primary-bold text-xl leading-9 hover:cursor-pointer sm:text-md text-quaternary border-b-4 border-quaternary"
        >
          Description
        </li>
        <li
          class="pb-2 font-primary-bold text-xl leading-9 hover:cursor-pointer sm:text-md text-tertiary"
        >
          Review (5)
        </li>
      </ul>
      <div
        class="mt-4"
      >
        <div
          class="text-md sm:text-sm"
        >
          <p
            class="text-secondary font-primary-bold"
          >
            Detail Product
          </p>
          <ul
            class="mt-5 ml-6 list-disc"
          >
            <li
              class="text-tertiary"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit
            </li>
            <li
              class="text-tertiary"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit
            </li>
            <li
              class="text-tertiary"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
